os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -68,6 +83,10 @@
 
 void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size_t size, int heap_type)
 {
+#ifdef __KERNEL__
```

Comment:
As my check, `__KERNEL__` is defined only when protected build.

Reply:
I think it is defined for flat also. 
`KDEFINE = ${shell $(TOPDIR)/tools/define.sh "$(CC)" __KERNEL__}`

It is called in pass2 which is executed for both flat and protected.

Reply:
I don't think so.
Flatlibs.mk
```
TINYARALIBS += ... $(LIBRARIES_DIR)$(DELIM)libmm$(LIBEXT)
```
LibTargets.mk
```
mm$(DELIM)libmm$(LIBEXT): context
	$(Q) $(MAKE) -C mm TOPDIR="$(TOPDIR)" libmm$(LIBEXT)

$(LIBRARIES_DIR)$(DELIM)libmm$(LIBEXT): mm$(DELIM)libmm$(LIBEXT)
	$(Q) install mm$(DELIM)libmm$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libmm$(LIBEXT)
```
There is no `KERNEL=y EXTRADEFINES=$(KDEFINE)`. When `KERNEL=y EXTRADEFINES=$(KDEFINE)` is given, CFLAGS has -D__KERNEL__.